packages/google_sign_in/google_sign_in_android/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -335,20 +339,28 @@ public void init(
             throw new IllegalStateException("Unknown signInOption");
         }
 
-        // Only requests a clientId if google-services.json was present and parsed
-        // by the google-services Gradle script.
-        // TODO(jackson): Perhaps we should provide a mechanism to override this
-        // behavior.
-        int clientIdIdentifier =
-            context
-                .getResources()
-                .getIdentifier("default_web_client_id", "string", context.getPackageName());
         if (!Strings.isNullOrEmpty(clientId)) {
-          optionsBuilder.requestIdToken(clientId);
-          optionsBuilder.requestServerAuthCode(clientId);
-        } else if (clientIdIdentifier != 0) {
-          optionsBuilder.requestIdToken(context.getString(clientIdIdentifier));
-          optionsBuilder.requestServerAuthCode(context.getString(clientIdIdentifier));
+          throw new IllegalArgumentException(
```

Comment:
Looks like the old behavior was using clientId as serverClientId. 
Can we avoid the breaking change by keep the old behavior for clientID? We can print a warning to guide devs to migrate to use the serverClientId.

Reply:
I couldn't find a documentation saying this is not supported, could you link me to it if you can?

Reply:
Done

Android apps don't have a client ID and are instead identified through their package name and the SHA-1 of their signing key.
https://developers.google.com/android/guides/client-auth
https://developers.google.com/identity/sign-in/android/start#configure-a-google-api-project